src/background/telemetry/metrics.js

```diff
@@ -352,7 +352,16 @@ export default class Metrics {
   _timeToExpired(type, frequency) {
     if (frequency === 'all') return 0;
 
-    const last = this.storage[`${type}_${frequency}`] || 0;
+    const key = `${type}_${frequency}`;
+
+    // Protect against calling events immediately after install for all frequencies
+    // They should trigger on the trailing edge of the frequency
+    if (!this.storage[key]) {
+      this.storage[key] = Date.now();
```

Comment:
since "Date.now()" is used on all paths (for frequency_ago), you can extract it.

Reply:
I hope I understood what you mean ;)